components/stimulation/StimulationStudioWaveformSettingModal.vue

```diff
@@ -383,31 +376,51 @@ export default {
       err_msg: {
         phase_one_duration: "",
         phase_one_charge: "",
-        interpulse_duration: "",
+        interphase_interval: "",
         phase_two_duration: "",
         phase_two_charge: "",
-        repeat_delay_interval: "",
+        pulse_frequency: "",
         total_active_duration: "",
       },
-      time_units: {
-        total_active_duration: ["milliseconds", "seconds"],
-        repeat_delay_interval: ["milliseconds", "seconds"],
-      },
+      time_units: ["milliseconds", "seconds", "minutes"],
       regex: {
         charge: new RegExp("^-?([0]{1}.{1}[0-9]+|[1-9]{1}[0-9]*.{1}[0-9]+|[0-9]+|0)$"),
         duration: new RegExp("^[0-9][0-9]*d*$"),
       },
       is_enabled_array: [false, true, true],
       all_valid: false,
-      delay_interval_idx: 0,
       active_duration_idx: 0,
+      input_pulse_frequency: 0,
     };
   },
   computed: {
-    // check_max_type: function() {
-    //   if (this.stimulation_type === "Current (mA)") return "Voltage (mV)";
-    //   else return "Current (mA)";
-    // }
+    total_pulse_duration: function () {
+      const { phase_one_duration, phase_two_duration, interphase_interval } = this.pulse_settings;
+      return Number(phase_one_duration) + Number(phase_two_duration) + Number(interphase_interval);
+    },
+    total_pulse_duration_on_open: function () {
+      const { phase_one_duration, phase_two_duration, interphase_interval } = this.selected_pulse_settings;
+      return Number(phase_one_duration) + Number(phase_two_duration) + Number(interphase_interval);
+    },
+    calculated_delay_on_close: function () {
+      const total_delay = 1000 - this.input_pulse_frequency * this.total_pulse_duration;
+      return total_delay / this.input_pulse_frequency;
+    },
+    calculated_frequency_on_open: function () {
+      const { repeat_delay_interval } = this.selected_stim_settings;
+      const one_cycle = repeat_delay_interval + this.total_pulse_duration_on_open;
+      let calculated_duration = 0;
+      let cycles = 0;
+
+      if (one_cycle == 0) return "";
+      else {
+        while (calculated_duration + one_cycle <= 1000) {
+          calculated_duration += one_cycle;
+          cycles++;
+        }
+        return cycles;
+      }
+    },
```

Comment:
Please let me know if this all looks correct or if there may be a better way to handle this. 
I wanted to do the conversion here so that it didn't affect our original protocol template to the BE. 

calculated_delay_on_close => Calculating the delay between each pulse given the total active duration and pulse parameters on close. 
calculated_frequency_on_open => Calculating the frequency on open for edit given the delay duration.